### DISCUSSION TITLE
Min/Max model constraints?

### DISCUSSION THREAD

**Discussion by cdeterman** [online-ml/river#1217] [Q&A] [answered]
Is there a way to set 'bounds/constraints' on the models?  For example, I have a forecast that can never be below 0.  Would there be a way internally to set such a constraint or should that be a post-processing step?  Modifying after the fact would be simple enough but I worry about the feedback to the internal 'coefficients' in subsequent runs in the model doesn't 'know' that the previous negative forecast was 'out of bounds'.

**Comment by MaxHalford** [comment, marked as answer]
Hey there. There is [`PredClipper`](https://riverml.xyz/0.15.0/api/preprocessing/PredClipper/) if you're too lazy to clip the predictions yourself.

> I worry about the feedback to the internal 'coefficients' in subsequent runs in the model doesn't 'know' that the previous negative forecast was 'out of bounds'.

You shouldn't worry about that. If your model is well behaved (e.g. a sane learning rate for a linear model) predictions will make sense, and won't be negative when they shouldn't be. In particular, if you take tree models, it's guaranteed they won't produce negative outputs if they've never seen any negative label.

**Reply by cdeterman** [reply]
Thank you for the clarification
